src/uct/tcp/tcp_ep.c

```diff
@@ -383,7 +383,9 @@ uct_tcp_ep_create_socket_and_connect(uct_tcp_iface_t *iface,
         goto err_ep_destroy;
     }
 
-    *ep_p = ep;
+    if (*ep_p == NULL) {
```

Comment:
move this to line 374?

Reply:
it wouldn't work, since we have check below (line 393) that it's not external EP (i.e. allocated here) and we should destroy it

Reply:
@yosefe is it ok?

Reply:
ok